protocol/consistencychecks.go

```diff
@@ -210,45 +223,106 @@ func (cc *ConsistencyChecks) verifySTR(str *m.SignedTreeRoot) error {
 	}
 
 	// verify hash chain
+	// TODO: clearly verify which is the client's actual expectation.
+	// See #81
 	if (str.Epoch == cc.CurrentEpoch && bytes.Equal(cc.SavedSTR, str.Signature)) ||
 		(str.Epoch == cc.CurrentEpoch+1 && str.VerifyHashChain(cc.SavedSTR)) {
 		return nil
 	}
 	return ErrorBadSTR
 }
 
-// verifyReturnedPromise verifies the returned TB
-// included in the directory's response.
-func (cc *ConsistencyChecks) verifyReturnedPromise(tb *m.TemporaryBinding,
-	uname string, key []byte,
-	str *m.SignedTreeRoot,
-	ap *m.AuthenticationPath) error {
+// verifySignedPromise checks if the returned response is match
+// with the directory's policy. Then it calls verifyFulfilledPromise
+// to verify whether issued TBs was inserted in the directory
+// as promised. Finally, it verifies the returned promise
```

Comment:
I think we should separate these two verifications since they really do have two different purposes. Checking whether a TB has been included usually only occurs during the very first monitoring check. Checking that the signed promise matches the registered binding only occurs after the registration. 

As for handling the TBs we receive for other users during lookups, we should decide if the client should verify them. For one, clients don't have enough information about the binding that *other* clients registered, so I'm not sure it makes sense to verify that the TB matches the expected binding. Clients could be cautious and verify that others' TBs have been included in the next epoch, but what does the client do if this verification fails?